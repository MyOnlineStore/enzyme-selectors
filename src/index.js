const Enzyme = require("enzyme");
const chalk = require("chalk");

function addSelectors() {
  function findByTestId(id) {
    return this.find(`[data-testid="${id}"]`).hostNodes();
  }

  function findByAriaLabel(label) {
    return this.find(`[aria-label="${label}"]`).hostNodes();
  }

  function findByPlaceholderText(text) {
    return this.find(`[aria-label="${text}"]`);
  }

  function findByAltText(text) {
    return this.find(`[alt="${text}"]`);
  }

  function findByTitle(title) {
    return this.find(`[title="${title}"]`);
  }

  function findByRole(role) {
    return this.find(`[role="${role}"]`);
  }

  function debugByAttribute(attr, value) {
    let report = "\n\n";
    let component = null;

    if (value) {
      component = this.find(`[${attr}="${value}"]`);
    } else {
      component = this.find(`[${attr}]`);
    }

    component.hostNodes().map(node => {
      const html = node.debug();
      const parts = html.split(value);

      const styled = parts.reduce((acc, part, index) => {
        if (index === parts.length - 1) {
          return `${acc}${chalk.dim(part)}`;
        }

        return `${acc}${chalk.dim(part)}${chalk.green(chalk.bold(value))}`;
      }, "");

      report += `${styled}\n\n`;
    });

    return report;
  }

  function debugByTestId(id) {
    return this.debugByAttribute("data-testid", id);
  }

  function debugByAriaLabel(label) {
    return this.debugByAttribute("aria-label", label);
  }

  function debugByPlaceholderText(text) {
    return this.debugByAttribute("placeholder", text);
  }

  function debugByAltText(text) {
    return this.debugByAttribute("alt", text);
  }

  function debugByTitle(title) {
    return this.debugByAttribute("title", title);
  }

  function debugByRole(role) {
    return this.debugByAttribute("role", role);
  }

  const applied = [
    debugByAltText,
    debugByAriaLabel,
    debugByAttribute,
    debugByPlaceholderText,
    debugByRole,
    debugByTestId,
    debugByTitle,
    findByAltText,
    findByAriaLabel,
    findByPlaceholderText,
    findByRole,
    findByTestId,
    findByTitle
  ];

  applied.forEach(item => {
    Enzyme.ReactWrapper.prototype[item.name] = item;
    Enzyme.ShallowWrapper.prototype[item.name] = item;
  });
}

module.exports = {
  addSelectors
};
