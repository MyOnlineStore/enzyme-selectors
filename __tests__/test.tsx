import React, { FC } from "react";
import { mount, shallow } from "enzyme";
import { addSelectors } from "../src";

const Foo: FC = props => {
  return <div {...props}>foo</div>;
};

describe("selectors", () => {
  describe.each([
    ["mount", mount],
    ["shallow", shallow]
  ])("%s", (_, render: typeof mount) => {
    it("should find a component by it's testid", () => {
      addSelectors();
      const component = render(<Foo data-testid="foo" />);
      expect(component.findByTestId("foo").length).toBe(1);
    });

    it("should find a component by it's alt text", () => {
      addSelectors();
      const component = render(<img alt="foo" />);
      expect(component.findByAltText("foo").length).toBe(1);
    });

    it("should find a component by it's aria-label", () => {
      addSelectors();
      const component = render(<Foo aria-label="foo" />);
      expect(component.findByAriaLabel("foo").length).toBe(1);
    });

    it("should find a component by it's placeholder", () => {
      addSelectors();
      const component = render(<input type="text" placeholder="foo" />);
      expect(component.findByPlaceholderText("foo").length).toBe(1);
    });

    it("should find a component by it's role", () => {
      addSelectors();
      const component = render(<button role="foo" />);
      expect(component.findByRole("foo").length).toBe(1);
    });

    it("should find a component by it's title", () => {
      addSelectors();
      const component = render(<a href="https://1234.com" title="foo" />);
      expect(component.findByTitle("foo").length).toBe(1);
    });
  });
});

describe("util", () => {
  describe.each([
    ["mount", mount],
    ["shallow", shallow]
  ])("%s", (_, render: typeof mount) => {
    describe.each([
      ["debugByAltText", "alt"],
      ["debugByAriaLabel", "aria-label"],
      ["debugByPlaceholderText", "placeholder"],
      ["debugByRole", "role"],
      ["debugByTestId", "data-testid"],
      ["debugByTitle", "title"]
    ])("%s", (fn: string, attr: string) => {
      addSelectors();
      it("should generate a debug report for all occurences", () => {
        const component = render(
          <div {...{ [attr]: "one" }}>
            <div {...{ [attr]: "two" }}>
              <div {...{ [attr]: "three" }}>
                <div {...{ [attr]: "four" }}>
                  <div {...{ [attr]: "double" }}></div>
                  <div {...{ [attr]: "double" }}></div>
                </div>
              </div>
            </div>
          </div>
        );

        expect(component[fn]()).toMatchSnapshot(`${render.name}:${fn}:all`);
      });

      it("should generate a debug report for all occurences", () => {
        const component = render(
          <div {...{ [attr]: "one" }}>
            <div {...{ [attr]: "two" }}>
              <div {...{ [attr]: "three" }}>
                <div {...{ [attr]: "four" }}>
                  <div {...{ [attr]: "double" }}></div>
                  <div {...{ [attr]: "double" }}></div>
                </div>
              </div>
            </div>
          </div>
        );

        expect(component[fn]("two")).toMatchSnapshot(
          `${render.name}:${fn}:single`
        );
      });
    });
  });
});
