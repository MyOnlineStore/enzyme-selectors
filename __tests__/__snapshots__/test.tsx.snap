// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util mount debugByAltText should generate a debug report for all occurences: mount:debugByAltText:all 1`] = `
"

[2m<div alt=\\"one\\">[22m
[2m  <div alt=\\"two\\">[22m
[2m    <div alt=\\"three\\">[22m
[2m      <div alt=\\"four\\">[22m
[2m        <div alt=\\"double\\" />[22m
[2m        <div alt=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div alt=\\"two\\">[22m
[2m  <div alt=\\"three\\">[22m
[2m    <div alt=\\"four\\">[22m
[2m      <div alt=\\"double\\" />[22m
[2m      <div alt=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div alt=\\"three\\">[22m
[2m  <div alt=\\"four\\">[22m
[2m    <div alt=\\"double\\" />[22m
[2m    <div alt=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div alt=\\"four\\">[22m
[2m  <div alt=\\"double\\" />[22m
[2m  <div alt=\\"double\\" />[22m
[2m</div>[22m

[2m<div alt=\\"double\\" />[22m

[2m<div alt=\\"double\\" />[22m

"
`;

exports[`util mount debugByAltText should generate a debug report for all occurences: mount:debugByAltText:single 1`] = `
"

[2m<div alt=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div alt=\\"three\\">[22m
[2m    <div alt=\\"four\\">[22m
[2m      <div alt=\\"double\\" />[22m
[2m      <div alt=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util mount debugByAriaLabel should generate a debug report for all occurences: mount:debugByAriaLabel:all 1`] = `
"

[2m<div aria-label=\\"one\\">[22m
[2m  <div aria-label=\\"two\\">[22m
[2m    <div aria-label=\\"three\\">[22m
[2m      <div aria-label=\\"four\\">[22m
[2m        <div aria-label=\\"double\\" />[22m
[2m        <div aria-label=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div aria-label=\\"two\\">[22m
[2m  <div aria-label=\\"three\\">[22m
[2m    <div aria-label=\\"four\\">[22m
[2m      <div aria-label=\\"double\\" />[22m
[2m      <div aria-label=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div aria-label=\\"three\\">[22m
[2m  <div aria-label=\\"four\\">[22m
[2m    <div aria-label=\\"double\\" />[22m
[2m    <div aria-label=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div aria-label=\\"four\\">[22m
[2m  <div aria-label=\\"double\\" />[22m
[2m  <div aria-label=\\"double\\" />[22m
[2m</div>[22m

[2m<div aria-label=\\"double\\" />[22m

[2m<div aria-label=\\"double\\" />[22m

"
`;

exports[`util mount debugByAriaLabel should generate a debug report for all occurences: mount:debugByAriaLabel:single 1`] = `
"

[2m<div aria-label=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div aria-label=\\"three\\">[22m
[2m    <div aria-label=\\"four\\">[22m
[2m      <div aria-label=\\"double\\" />[22m
[2m      <div aria-label=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util mount debugByPlaceholderText should generate a debug report for all occurences: mount:debugByPlaceholderText:all 1`] = `
"

[2m<div placeholder=\\"one\\">[22m
[2m  <div placeholder=\\"two\\">[22m
[2m    <div placeholder=\\"three\\">[22m
[2m      <div placeholder=\\"four\\">[22m
[2m        <div placeholder=\\"double\\" />[22m
[2m        <div placeholder=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div placeholder=\\"two\\">[22m
[2m  <div placeholder=\\"three\\">[22m
[2m    <div placeholder=\\"four\\">[22m
[2m      <div placeholder=\\"double\\" />[22m
[2m      <div placeholder=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div placeholder=\\"three\\">[22m
[2m  <div placeholder=\\"four\\">[22m
[2m    <div placeholder=\\"double\\" />[22m
[2m    <div placeholder=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div placeholder=\\"four\\">[22m
[2m  <div placeholder=\\"double\\" />[22m
[2m  <div placeholder=\\"double\\" />[22m
[2m</div>[22m

[2m<div placeholder=\\"double\\" />[22m

[2m<div placeholder=\\"double\\" />[22m

"
`;

exports[`util mount debugByPlaceholderText should generate a debug report for all occurences: mount:debugByPlaceholderText:single 1`] = `
"

[2m<div placeholder=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div placeholder=\\"three\\">[22m
[2m    <div placeholder=\\"four\\">[22m
[2m      <div placeholder=\\"double\\" />[22m
[2m      <div placeholder=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util mount debugByRole should generate a debug report for all occurences: mount:debugByRole:all 1`] = `
"

[2m<div role=\\"one\\">[22m
[2m  <div role=\\"two\\">[22m
[2m    <div role=\\"three\\">[22m
[2m      <div role=\\"four\\">[22m
[2m        <div role=\\"double\\" />[22m
[2m        <div role=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div role=\\"two\\">[22m
[2m  <div role=\\"three\\">[22m
[2m    <div role=\\"four\\">[22m
[2m      <div role=\\"double\\" />[22m
[2m      <div role=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div role=\\"three\\">[22m
[2m  <div role=\\"four\\">[22m
[2m    <div role=\\"double\\" />[22m
[2m    <div role=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div role=\\"four\\">[22m
[2m  <div role=\\"double\\" />[22m
[2m  <div role=\\"double\\" />[22m
[2m</div>[22m

[2m<div role=\\"double\\" />[22m

[2m<div role=\\"double\\" />[22m

"
`;

exports[`util mount debugByRole should generate a debug report for all occurences: mount:debugByRole:single 1`] = `
"

[2m<div role=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div role=\\"three\\">[22m
[2m    <div role=\\"four\\">[22m
[2m      <div role=\\"double\\" />[22m
[2m      <div role=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util mount debugByTestId should generate a debug report for all occurences: mount:debugByTestId:all 1`] = `
"

[2m<div data-testid=\\"one\\">[22m
[2m  <div data-testid=\\"two\\">[22m
[2m    <div data-testid=\\"three\\">[22m
[2m      <div data-testid=\\"four\\">[22m
[2m        <div data-testid=\\"double\\" />[22m
[2m        <div data-testid=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div data-testid=\\"two\\">[22m
[2m  <div data-testid=\\"three\\">[22m
[2m    <div data-testid=\\"four\\">[22m
[2m      <div data-testid=\\"double\\" />[22m
[2m      <div data-testid=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div data-testid=\\"three\\">[22m
[2m  <div data-testid=\\"four\\">[22m
[2m    <div data-testid=\\"double\\" />[22m
[2m    <div data-testid=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div data-testid=\\"four\\">[22m
[2m  <div data-testid=\\"double\\" />[22m
[2m  <div data-testid=\\"double\\" />[22m
[2m</div>[22m

[2m<div data-testid=\\"double\\" />[22m

[2m<div data-testid=\\"double\\" />[22m

"
`;

exports[`util mount debugByTestId should generate a debug report for all occurences: mount:debugByTestId:single 1`] = `
"

[2m<div data-testid=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div data-testid=\\"three\\">[22m
[2m    <div data-testid=\\"four\\">[22m
[2m      <div data-testid=\\"double\\" />[22m
[2m      <div data-testid=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util mount debugByTitle should generate a debug report for all occurences: mount:debugByTitle:all 1`] = `
"

[2m<div title=\\"one\\">[22m
[2m  <div title=\\"two\\">[22m
[2m    <div title=\\"three\\">[22m
[2m      <div title=\\"four\\">[22m
[2m        <div title=\\"double\\" />[22m
[2m        <div title=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div title=\\"two\\">[22m
[2m  <div title=\\"three\\">[22m
[2m    <div title=\\"four\\">[22m
[2m      <div title=\\"double\\" />[22m
[2m      <div title=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div title=\\"three\\">[22m
[2m  <div title=\\"four\\">[22m
[2m    <div title=\\"double\\" />[22m
[2m    <div title=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div title=\\"four\\">[22m
[2m  <div title=\\"double\\" />[22m
[2m  <div title=\\"double\\" />[22m
[2m</div>[22m

[2m<div title=\\"double\\" />[22m

[2m<div title=\\"double\\" />[22m

"
`;

exports[`util mount debugByTitle should generate a debug report for all occurences: mount:debugByTitle:single 1`] = `
"

[2m<div title=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div title=\\"three\\">[22m
[2m    <div title=\\"four\\">[22m
[2m      <div title=\\"double\\" />[22m
[2m      <div title=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util shallow debugByAltText should generate a debug report for all occurences: shallow:debugByAltText:all 1`] = `
"

[2m<div alt=\\"one\\">[22m
[2m  <div alt=\\"two\\">[22m
[2m    <div alt=\\"three\\">[22m
[2m      <div alt=\\"four\\">[22m
[2m        <div alt=\\"double\\" />[22m
[2m        <div alt=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div alt=\\"two\\">[22m
[2m  <div alt=\\"three\\">[22m
[2m    <div alt=\\"four\\">[22m
[2m      <div alt=\\"double\\" />[22m
[2m      <div alt=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div alt=\\"three\\">[22m
[2m  <div alt=\\"four\\">[22m
[2m    <div alt=\\"double\\" />[22m
[2m    <div alt=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div alt=\\"four\\">[22m
[2m  <div alt=\\"double\\" />[22m
[2m  <div alt=\\"double\\" />[22m
[2m</div>[22m

[2m<div alt=\\"double\\" />[22m

[2m<div alt=\\"double\\" />[22m

"
`;

exports[`util shallow debugByAltText should generate a debug report for all occurences: shallow:debugByAltText:single 1`] = `
"

[2m<div alt=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div alt=\\"three\\">[22m
[2m    <div alt=\\"four\\">[22m
[2m      <div alt=\\"double\\" />[22m
[2m      <div alt=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util shallow debugByAriaLabel should generate a debug report for all occurences: shallow:debugByAriaLabel:all 1`] = `
"

[2m<div aria-label=\\"one\\">[22m
[2m  <div aria-label=\\"two\\">[22m
[2m    <div aria-label=\\"three\\">[22m
[2m      <div aria-label=\\"four\\">[22m
[2m        <div aria-label=\\"double\\" />[22m
[2m        <div aria-label=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div aria-label=\\"two\\">[22m
[2m  <div aria-label=\\"three\\">[22m
[2m    <div aria-label=\\"four\\">[22m
[2m      <div aria-label=\\"double\\" />[22m
[2m      <div aria-label=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div aria-label=\\"three\\">[22m
[2m  <div aria-label=\\"four\\">[22m
[2m    <div aria-label=\\"double\\" />[22m
[2m    <div aria-label=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div aria-label=\\"four\\">[22m
[2m  <div aria-label=\\"double\\" />[22m
[2m  <div aria-label=\\"double\\" />[22m
[2m</div>[22m

[2m<div aria-label=\\"double\\" />[22m

[2m<div aria-label=\\"double\\" />[22m

"
`;

exports[`util shallow debugByAriaLabel should generate a debug report for all occurences: shallow:debugByAriaLabel:single 1`] = `
"

[2m<div aria-label=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div aria-label=\\"three\\">[22m
[2m    <div aria-label=\\"four\\">[22m
[2m      <div aria-label=\\"double\\" />[22m
[2m      <div aria-label=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util shallow debugByPlaceholderText should generate a debug report for all occurences: shallow:debugByPlaceholderText:all 1`] = `
"

[2m<div placeholder=\\"one\\">[22m
[2m  <div placeholder=\\"two\\">[22m
[2m    <div placeholder=\\"three\\">[22m
[2m      <div placeholder=\\"four\\">[22m
[2m        <div placeholder=\\"double\\" />[22m
[2m        <div placeholder=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div placeholder=\\"two\\">[22m
[2m  <div placeholder=\\"three\\">[22m
[2m    <div placeholder=\\"four\\">[22m
[2m      <div placeholder=\\"double\\" />[22m
[2m      <div placeholder=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div placeholder=\\"three\\">[22m
[2m  <div placeholder=\\"four\\">[22m
[2m    <div placeholder=\\"double\\" />[22m
[2m    <div placeholder=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div placeholder=\\"four\\">[22m
[2m  <div placeholder=\\"double\\" />[22m
[2m  <div placeholder=\\"double\\" />[22m
[2m</div>[22m

[2m<div placeholder=\\"double\\" />[22m

[2m<div placeholder=\\"double\\" />[22m

"
`;

exports[`util shallow debugByPlaceholderText should generate a debug report for all occurences: shallow:debugByPlaceholderText:single 1`] = `
"

[2m<div placeholder=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div placeholder=\\"three\\">[22m
[2m    <div placeholder=\\"four\\">[22m
[2m      <div placeholder=\\"double\\" />[22m
[2m      <div placeholder=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util shallow debugByRole should generate a debug report for all occurences: shallow:debugByRole:all 1`] = `
"

[2m<div role=\\"one\\">[22m
[2m  <div role=\\"two\\">[22m
[2m    <div role=\\"three\\">[22m
[2m      <div role=\\"four\\">[22m
[2m        <div role=\\"double\\" />[22m
[2m        <div role=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div role=\\"two\\">[22m
[2m  <div role=\\"three\\">[22m
[2m    <div role=\\"four\\">[22m
[2m      <div role=\\"double\\" />[22m
[2m      <div role=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div role=\\"three\\">[22m
[2m  <div role=\\"four\\">[22m
[2m    <div role=\\"double\\" />[22m
[2m    <div role=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div role=\\"four\\">[22m
[2m  <div role=\\"double\\" />[22m
[2m  <div role=\\"double\\" />[22m
[2m</div>[22m

[2m<div role=\\"double\\" />[22m

[2m<div role=\\"double\\" />[22m

"
`;

exports[`util shallow debugByRole should generate a debug report for all occurences: shallow:debugByRole:single 1`] = `
"

[2m<div role=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div role=\\"three\\">[22m
[2m    <div role=\\"four\\">[22m
[2m      <div role=\\"double\\" />[22m
[2m      <div role=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util shallow debugByTestId should generate a debug report for all occurences: shallow:debugByTestId:all 1`] = `
"

[2m<div data-testid=\\"one\\">[22m
[2m  <div data-testid=\\"two\\">[22m
[2m    <div data-testid=\\"three\\">[22m
[2m      <div data-testid=\\"four\\">[22m
[2m        <div data-testid=\\"double\\" />[22m
[2m        <div data-testid=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div data-testid=\\"two\\">[22m
[2m  <div data-testid=\\"three\\">[22m
[2m    <div data-testid=\\"four\\">[22m
[2m      <div data-testid=\\"double\\" />[22m
[2m      <div data-testid=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div data-testid=\\"three\\">[22m
[2m  <div data-testid=\\"four\\">[22m
[2m    <div data-testid=\\"double\\" />[22m
[2m    <div data-testid=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div data-testid=\\"four\\">[22m
[2m  <div data-testid=\\"double\\" />[22m
[2m  <div data-testid=\\"double\\" />[22m
[2m</div>[22m

[2m<div data-testid=\\"double\\" />[22m

[2m<div data-testid=\\"double\\" />[22m

"
`;

exports[`util shallow debugByTestId should generate a debug report for all occurences: shallow:debugByTestId:single 1`] = `
"

[2m<div data-testid=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div data-testid=\\"three\\">[22m
[2m    <div data-testid=\\"four\\">[22m
[2m      <div data-testid=\\"double\\" />[22m
[2m      <div data-testid=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;

exports[`util shallow debugByTitle should generate a debug report for all occurences: shallow:debugByTitle:all 1`] = `
"

[2m<div title=\\"one\\">[22m
[2m  <div title=\\"two\\">[22m
[2m    <div title=\\"three\\">[22m
[2m      <div title=\\"four\\">[22m
[2m        <div title=\\"double\\" />[22m
[2m        <div title=\\"double\\" />[22m
[2m      </div>[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div title=\\"two\\">[22m
[2m  <div title=\\"three\\">[22m
[2m    <div title=\\"four\\">[22m
[2m      <div title=\\"double\\" />[22m
[2m      <div title=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div title=\\"three\\">[22m
[2m  <div title=\\"four\\">[22m
[2m    <div title=\\"double\\" />[22m
[2m    <div title=\\"double\\" />[22m
[2m  </div>[22m
[2m</div>[22m

[2m<div title=\\"four\\">[22m
[2m  <div title=\\"double\\" />[22m
[2m  <div title=\\"double\\" />[22m
[2m</div>[22m

[2m<div title=\\"double\\" />[22m

[2m<div title=\\"double\\" />[22m

"
`;

exports[`util shallow debugByTitle should generate a debug report for all occurences: shallow:debugByTitle:single 1`] = `
"

[2m<div title=\\"[22m[32m[1mtwo[22m[39m[2m\\">[22m
[2m  <div title=\\"three\\">[22m
[2m    <div title=\\"four\\">[22m
[2m      <div title=\\"double\\" />[22m
[2m      <div title=\\"double\\" />[22m
[2m    </div>[22m
[2m  </div>[22m
[2m</div>[22m

"
`;
